src/Patterns/PatternImages.php

```diff
@@ -29,6 +29,77 @@ public function get_patterns_dictionary() {
 		return wp_json_file_decode( $patterns_dictionary, array( 'associative' => true ) );
 	}
 
+	/**
+	 * Returns the pattern images.
+	 *
+	 * @param string $pattern_slug The pattern slug.
+	 *
+	 * @return array The pattern images.
+	 */
+	public static function get_pattern_images( string $pattern_slug ): array {
+		$dictionary = get_option( self::WC_BLOCKS_PATTERNS_CONTENT );
+		if ( empty( $dictionary ) ) {
+			return array();
+		}
+
+		$pattern = null;
+		foreach ( $dictionary as $item ) {
+			if ( $item['slug'] === $pattern_slug ) {
+				$pattern = $item;
+				break;
+			}
+		}
+
+		if ( empty( $pattern ) ) {
+			return array();
+		}
+
+		if ( ! isset( $pattern['images'] ) ) {
+			return array();
+		}
+
+		if ( ! isset( $pattern['images_total'] ) ) {
+			return array();
+		}
+
+		return self::get_random_images( $pattern['images'], $pattern['images_total'] );
+	}
+
+	/**
+	 * Returns the image for the given pattern.
+	 *
+	 * @param array  $images The array of images.
+	 * @param int    $index The index of the image to return.
+	 * @param string $default_image The default image to return.
+	 *
+	 * @return string The image.
+	 */
+	public static function get_image_url( array $images, int $index, string $default_image ): string {
```

Comment:
this also seems like a unit test would be nice.

Reply:
Same https://github.com/woocommerce/woocommerce-blocks/pull/10796/files#r1328597336